### ISSUE TITLE
Critical Error, SNMP stops working when monitoring with PRTG

### ISSUE THREAD

**Issue by thebrainiak** [net-snmp/net-snmp#601]
Hi, i am using the last version of snmp

![image](https://github.com/net-snmp/net-snmp/assets/101867702/c56e29b3-9f57-4d2d-a334-83ddc6c27855)

I recieve a core-dump error each day and snmp stops working suddenly, only starts to work when i restart it.

![Captura de pantalla 2023-08-01 140740](https://github.com/net-snmp/net-snmp/assets/101867702/42c306e4-4098-4e0f-b599-42209d853c19)

This is my configuration file: 

###########################################################################
#
# snmpd.conf
# An example configuration file for configuring the Net-SNMP agent ('snmpd')
# See snmpd.conf(5) man page for details
#
###########################################################################
# SECTION: System Information Setup
#

# syslocation: The [typically physical] location of the system.
#   Note that setting this value here means that when trying to
#   perform an snmp SET operation to the sysLocation.0 variable will make
#   the agent return the "notWritable" error code.  IE, including
#   this token in the snmpd.conf file will disable write access to
#   the variable.
#   arguments:  location_string
sysLocation    Sitting on the Dock of the Bay
sysContact     Me <me@example.org>

# sysservices: The proper value for the sysServices object.
#   arguments:  sysservices_number
sysServices    72

###########################################################################
# SECTION: Agent Operating Mode
#
#   This section defines how the agent will operate when it
#   is running.

# master: Should the agent operate as a master agent or not.
#   Currently, the only supported master agent type for this token
#   is "agentx".
#
#   arguments: (on|yes|agentx|all|off|no)

master  agentx

# agentaddress: The IP address and port number that the agent will listen on.
#   By default the agent listens to any and all traffic from any
#   interface on the default SNMP port (161).  This allows you to
#   specify which address, interface, transport type and port(s) that you
#   want the agent to listen on.  Multiple definitions of this token
#   are concatenated together (using ':'s).
#   arguments: [transport:]port[@interface/address],...
#agentaddress  udp:127.0.0.1:161
#agentaddress  udp:161,udp6:[::1]:161
#agentaddress   192.168.XXX.XXX:161
agentAddress udp:127.0.0.1:161,udp:192.168.XXX.XXX:161

###########################################################################
# SECTION: Access Control Setup
#
#   This section defines who is allowed to talk to your running
#   snmp agent.

# Views
#   arguments viewname included [oid]

#  system + hrSystem groups only
##view   systemonly  included   .1.3.6.1.2.1.1
##view   systemonly  included   .1.3.6.1.2.1.25.1

# rocommunity: a SNMPv1/SNMPv2c read-only access community name
#   arguments:  community [default|hostname|network/bits] [oid | -V view]

# Read-only access to everyone to the systemonly view
rocommunity  XXXX 192.168.XXX.XX
#rocommunity6 public default -V systemonly
#rocommunity XXXX

# SNMPv3 doesn't use communities, but users with (optionally) an
# authentication and encryption string. This user needs to be created
# with what they can view with rouser/rwuser lines in this file.
#
# createUser username (MD5|SHA|SHA-512|SHA-384|SHA-256|SHA-224) authpassphrase [DES|AES] [privpassphrase]
# e.g.
# createuser authPrivUser SHA-512 myauthphrase AES myprivphrase
#
# This should be put into /var/lib/snmp/snmpd.conf
#
# rouser: a SNMPv3 read-only access username
#    arguments: username [noauth|auth|priv [OID | -V VIEW [CONTEXT]]]
rouser authPrivUser authpriv -V systemonly

# include a all *.conf files in a directory
includeDir /etc/snmp/snmpd.conf.d

I was researching in the issues of this github and i saw that the solution could be to give more space to the buffer, i tried it with the snmp.conf file that is this:

# As the snmp packages come without MIB files due to license reasons, loading
# of MIBs is disabled by default. If you added the MIBs you can reenable
# loading them by commenting out the following line.
mibs :

# If you want to globally change where snmp libraries, commands and daemons
# look for MIBS, change the line below. Note you can set this for individual
# tools with the -M option or MIBDIRS environment variable.
#
# mibdirs /usr/share/snmp/mibs:/usr/share/snmp/mibs/iana:/usr/share/snmp/mibs/ietf

mbdirs /usr/share/snmp/mibs:/usr/share/snmp/mibs/iana:/usr/share/snmp/mibs/ietf:/usr/share/mibs/site:/usr/share/snmp/mibs:/usr/share/mibs/iana:/usr/share/mibs/ietf:/usr/share/mibs/netsnmp:20480

But still Crashing, I will be very glad with your team if you can help me. Thank you and best regards! Alex.

**Comment by nielsb** [comment]
First of all, the current version is 5.9.3, and has been for a couple of years, so it might be worth while to upgrade. You might even try the current 5.9.4.rc1 :-)
Second, what OS is this?

**Comment by thebrainiak** [comment]
Okey, and where can i find the 5.9.3 version or the 5.9.4? I dont find it anywhere and the compilation of the github repository gives me errors everywhere. Im using Ubuntu Server 22.04 . Thank you!

**Comment by nielsb** [comment]
Strange. I routinely compile and test on Ubuntu, FreeBSD 12/13, Debian 10/11/12, Fedora 36/37/38 and more, without problems. What specific errors do you see?

**Comment by hardaker** [comment]
https://sourceforge.net/projects/net-snmp/files/net-snmp/5.9.4-pre-releases/

**Comment by thebrainiak** [comment]
> Strange. I routinely compile and test on Ubuntu, FreeBSD 12/13, Debian 10/11/12, Fedora 36/37/38 and more, without problems. What specific errors do you see?

Check the issue: https://github.com/net-snmp/net-snmp/issues/604  Thank you!

**Comment by thebrainiak** [comment]
> https://sourceforge.net/projects/net-snmp/files/net-snmp/5.9.4-pre-releases/

Yes, but when i compile it and install it i get the 5.9.1 version again, this is the issue https://github.com/net-snmp/net-snmp/issues/604
